### PR TITLE
Support for API Version 800/1000/1200 for Hypervisor Cluster Profile

### PR DESCRIPTION
**Pull request by chebroluharika**
**Description**
Support for API Version 800/1000/1200 for Hypervisor Cluster Profile

**Issues Resolved**
NA

### Check List
- [ ] New functionality includes testing.
  - [x] All tests pass (`$ rake test`).
- [ ] New functionality has been documented in the README if applicable.
  - [ ] New functionality has been thoroughly documented in the examples (please include helpful comments).
- [x] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,34 @@
-# v2.4.1 (unreleased)
+# v2.5.0 (unreleased)
+### Notes
+- This release supports OneView Rest API versions 800/1000/1200 for Hypervisor Resources.
+
+### Oneview Features supported
```
Comment (soodpr) [inline CHANGELOG.md:6]: Please add only the resource which is new supported for this release. ie. Hypervisor Cluster Profile in this case

File: examples/hypervisor_cluster_profile.pp [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+# NOTE: As with all resources, the found ensurable accepts a data as an optional filter field.
+
+# This created Hypervisor ClusterProfile if we use ServerProfileTemplate with OS DeploymentPlan
```
Comment (soodpr) [inline examples/hypervisor_cluster_profile.pp]: below example does not contain the deployment plan.

File: examples/hypervisor_cluster_profile.pp [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+      path                          => 'DC2',
+      hypervisorType                => 'Vmware',
+      hypervisorHostProfileTemplate => {
+      serverProfileTemplateUri => '/rest/server-profile-templates/278cadfb-2e86-4a05-8932-972553518259',
+      hostprefix               => 'Test-Cluster-host'
+      }
+    }
+}
+
+# This creates Hypervisor Cluster Profile using Server Profile Template without OSDeploymentPlans
```
Comment (soodpr) [inline examples/hypervisor_cluster_profile.pp]: i think you have interchange the comments here

File: spec/unit/provider/oneview_hypervisor_cluster_profile_c7000_spec.rb [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+require 'spec_helper'
+
```
Comment (soodpr) [inline spec/unit/provider/oneview_hypervisor_cluster_profile_c7000_spec.rb:18]: import of shared_context is missing here

File: spec/unit/provider/oneview_hypervisor_cluster_profile_c7000_spec.rb [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+provider_class = Puppet::Type.type(:oneview_hypervisor_cluster_profile).provider(:c7000)
+api_version = 800
```
Comment (soodpr) [inline spec/unit/provider/oneview_hypervisor_cluster_profile_c7000_spec.rb:20]: no need to specify api version here if shared context is imported

File: spec/unit/provider/oneview_hypervisor_cluster_profile_c7000_spec.rb (hunk 4 above)
Comment (chebroluharika) [reply spec/unit/provider/oneview_hypervisor_cluster_profile_c7000_spec.rb:18]: We are importing shared_context using include_context.

Comment (soodpr) [reply spec/unit/provider/oneview_hypervisor_cluster_profile_c7000_spec.rb:18]: sorry missed it.. please ignore

File: examples/hypervisor_cluster_profile.pp (hunk 3 above)
Comment (chebroluharika) [reply examples/hypervisor_cluster_profile.pp]: done

File: examples/hypervisor_cluster_profile.pp (hunk 2 above)
Comment (chebroluharika) [reply examples/hypervisor_cluster_profile.pp]: updated comment accordingly

File: CHANGELOG.md (hunk 1 above)
Comment (chebroluharika) [reply CHANGELOG.md:6]: done

**Review by soodpr — APPROVED** [review]
LGTM 👍

**Review by jyothisgm — APPROVED** [review]
LGTM

**Pull request merged**